src/control/server/harness.go

```diff
@@ -156,44 +158,61 @@ func (h *EngineHarness) OnDrpcFailure(fns ...func(ctx context.Context, err error
 
 // CallDrpc calls the supplied dRPC method on a managed I/O Engine instance.
 func (h *EngineHarness) CallDrpc(ctx context.Context, method drpc.Method, body proto.Message) (resp *drpc.Response, err error) {
+	instances := h.Instances()
+	drpcErrs := make([]error, 0, len(instances))
 	defer func() {
 		if err == nil {
-			return
+			return // At least one engine was accessed.
 		}
-		// If the context was canceled, don't trigger callbacks.
-		if errors.Cause(err) == context.Canceled {
-			return
+
+		var e error
+		hasDRPCErr := false
+		for _, e = range drpcErrs {
+			switch e {
+			case errDRPCNotReady, FaultDataPlaneNotStarted:
```

Comment:
Please could you tell me why `errEngineNotReady` is not included as it is not a DPC a error.
From what I understand, it should be more  `errDRPCNotReady` which should not be included (as it was before).
However, I am not sure if such error should be managed by `onDrpcFailure` as it seems to not really be a DRPC error.

Reply:
This line describes errors that should trigger a leadership election which is needed when engines can no longer can be communicated with. `errEngineNotReady` is not in this list as it's typically associated with start-up related temporal un-availability. `errEngineNotReady` is in the retry list however.

Reply:
OK, thanks for the explanation.